labgrid/driver/powerdriver.py

```diff
@@ -279,3 +279,49 @@ def get(self):
             if b"off" in status:
                 return False
         raise ExecutionError("Did not find port status in uhubctl output ({})".format(repr(output)))
+
+
+@target_factory.reg_driver
+@attr.s(eq=False)
+class PDUDaemonDriver(Driver, PowerResetMixin, PowerProtocol):
+    """PDUDaemonDriver - Driver using a PDU port available via pdudaemon"""
+    bindings = {"port": "PDUDaemonPort", }
+    delay = attr.ib(default=5, validator=attr.validators.instance_of(int))
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+        self._requests = import_module('requests')
+        self._host = None
+        self._port = None
+
+    def _build_url(self, cmd):
+        res = "http://{}:{}/power/control/{}?hostname={}&port={}".format(
+            self._host, self._port, cmd, self.port.pdu, self.port.index)
+        if cmd == 'reboot':
+            res += "&delay={}".format(self.delay)
+        return res
+
+    def on_activate(self):
+        self._host, self._port = proxymanager.get_host_and_port(self.port, default_port=16421)
+
+    @Driver.check_active
+    @step()
+    def on(self):
+        self._requests.get(self._build_url('on'))
```

Comment:
According to the README, pdudaemon can return 503 as a status if the request is not accepted, we should raise an error/exception if this is the case.

Reply:
requests will raise an exception in that case. ;)

Reply:
iPyhton disagrees:
```
In [4]: requests.get("https://httpstat.us/503")  
Out[4]: <Response [503]>

In [5]:                                          
```
According to the [docs](https://requests.kennethreitz.org/en/master/user/quickstart/#response-status-codes) `raise_for_status()` needs to be called on the response object.

Reply:
Thanks! I've fixed this and updated the PR.